### ISSUE TITLE
Print title and orientation issue on firefox

### ISSUE THREAD

**Issue by NumanArshad** [MatthewHerbst/react-to-print#391]
In mozilla firefox print title and page landscape orientation is not working while It's working in microsoft edge and google chrome
Here are screenshot for chrome, edge and mozella
![Chrome](https://user-images.githubusercontent.com/37812712/123745741-22933c00-d8ca-11eb-9fc4-5e37ed6d2e43.PNG)
![Edge](https://user-images.githubusercontent.com/37812712/123745751-245cff80-d8ca-11eb-9a00-2eddfcffb4f6.PNG)
![Mozella_FireFox](https://user-images.githubusercontent.com/37812712/123746776-85d19e00-d8cb-11eb-9de0-4c49f6395845.PNG)

and here is my code
![code](https://user-images.githubusercontent.com/37812712/123746801-8ff39c80-d8cb-11eb-954f-4c69d80fb919.PNG)

**Comment by MatthewHerbst** [comment]
The title not working is being tracked in #321 

Any chance you can make a working codesandbox with your code that replicates the orientation issue? Everything in your code above looks ok but hard to debug/play around with an image

**Comment by NumanArshad** [comment]
Here is codesandbox url, I want to use lg grid system which is working on chrome, edge but not in firefox
but in codesandbox it's working even in firefox but not locally 
https://codesandbox.io/s/intelligent-galileo-g2z1g?file=/src/App.js

**Comment by NumanArshad** [comment]
Another query related orientation is , how to set default orientation landscape in firefox as @page {size : landscape } is not supported in firefox?

**Comment by MatthewHerbst** [comment]
Apologies for not responding, I've had to focus on work the past few weeks. I hope to have time to get to these issues this weekend

**Comment by pgarr** [comment]
I have similar issue with Firefox, that regardless of documentTitle prop or explicity setting document.title, title displayed in "save as" window is always "blank.pdf".
I made some research and read this somewhere (can't find source now):
Chrome takes default title from most upper html tag, but Firefox from the most bottom one.

As react-to-print creates iframe for printing, I assumed, that Firefox takes title from html inside this iframe.
So I made fast and simple test to check this out:

    const handlePrint = useReactToPrint({
        content: () => printContainer.current,
        onBeforePrint: () => {
            const iframe: any = document.getElementById('printWindow');
            iframe!.contentDocument.title  = "iframe innerHtml title";
        },
        documentTitle: "default document title"
    });

With this code, in Chrome I have title "default document title", but in Firefox, title is "iframe innerHtml title".
So I think that setting this title for html inside iframe during its creation would fix this one issue.

**Comment by MatthewHerbst** [comment]
Thanks for the research @pgarr!

**Comment by MatthewHerbst** [comment]
After a lot of research and trial/error I've got the page title working for the major desktop browsers (Chrome, Firefox, Safari). I'll be making a release tonight with these changes, here is what it ended up needing to be:

```js
if (documentTitle) {
  // Print filename in Chrome
  target.ownerDocument.title = documentTitle;

  // Print filename in Firefox, Safari
  if (target.contentDocument) {
    target.contentDocument.title = documentTitle;
  }
}
```

I'll be doing research into mobile browsers later this week, but I believe the above should fix the majority use cases.

**Comment by MatthewHerbst** [comment]
Published in `v2.14.0`. Please let me know if you run into this again, and thanks for the help everyone
